src/Fields/interpolate.jl

```diff
@@ -49,37 +53,59 @@ end
     end
 end
 
-####
-#### Disclaimer! interpolation on LatitudeLongitudeGrid assumes a thin shell (i.e. no curvature effects when interpolating)
-#### Use other methods if a more accurate interpolation is required
-####
-
-@inline fractional_x_index(x::FT, ::Nothing, grid) where FT = one(FT)
-@inline fractional_y_index(y::FT, ::Nothing, grid) where FT = one(FT)
-@inline fractional_z_index(z::FT, ::Nothing, grid) where FT = one(FT)
-
-@inline fractional_x_index(x::FT, ::Center, grid) where FT = fractional_index(length(Center(), topology(grid, 1)(), size(grid, 1)), x, xnodes(grid, Center()))
-@inline fractional_y_index(y::FT, ::Center, grid) where FT = fractional_index(length(Center(), topology(grid, 2)(), size(grid, 2)), y, ynodes(grid, Center()))
-@inline fractional_z_index(z::FT, ::Center, grid) where FT = fractional_index(length(Center(), topology(grid, 3)(), size(grid, 3)), z, znodes(grid, Center()))
-
-@inline fractional_x_index(x::FT, ::Face, grid) where FT = fractional_index(length(Face(), topology(grid, 1)(), size(grid, 1)), x, xnodes(grid, Face())) - 1
-@inline fractional_y_index(y::FT, ::Face, grid) where FT = fractional_index(length(Face(), topology(grid, 2)(), size(grid, 2)), y, ynodes(grid, Face())) - 1
-@inline fractional_z_index(z::FT, ::Face, grid) where FT = fractional_index(length(Face(), topology(grid, 3)(), size(grid, 3)), z, znodes(grid, Face())) - 1
+#####
+##### Disclaimer! interpolation on LatitudeLongitudeGrid assumes a thin shell (i.e. no curvature effects when interpolating)
+##### Use other methods if a more accurate interpolation is required
+#####
+
+@inline function fractional_x_index(x::FT, locs, grid) where {FT}
```

Comment:
I dont think `FT` is used in this function. Also it might be better to use `x = nodes(grid, locs)[1]` and change the name of the input to something like `xi`.

Reply:
true, I missed that!